ambry-router/src/main/java/com/github/ambry/router/TtlUpdateOperation.java

```diff
@@ -388,6 +403,61 @@ private void checkAndMaybeComplete() {
     }
   }
 
+  /**
+   * Try to replicate the blob and then retry the TtlUpdate Request
+   * @return true if it's under retry. return false if cannot do retry or retry is finished.
+   */
+  private boolean retryWithReplicateBlob() {
+    // the conditions to trigger ReplicateBlob operation:
+    // 1. at least one replica returned successful status
+    // 2. and at least one replica returned NOT_FOUND status.
+    if (!operationTracker.hasNotFound() || operationTracker.getSuccessReplica().size() == 0) {
```

Comment:
we can put this if statement out side of retry method like my snippet above.

Reply:
Moved this condition out to the caller.